src/Compilers/CSharp/Portable/Errors/CSharpDiagnosticFormatter.cs

```diff
@@ -1,15 +1,29 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System.Globalization;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Diagnostics.CSharp;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
     public class CSharpDiagnosticFormatter : DiagnosticFormatter
     {
-        internal CSharpDiagnosticFormatter()
+        private static readonly string CompilerAnalyzer = typeof(CSharpCompilerDiagnosticAnalyzer).ToString();
+
+        internal CSharpDiagnosticFormatter(bool displayAnalyzer) :
+            base(displayAnalyzer)
         {
         }
 
-        public new static readonly CSharpDiagnosticFormatter Instance = new CSharpDiagnosticFormatter();
+        protected override string GetAnalyzerIdentity(DiagnosticAnalyzer analyzer)
```

Comment:
ya, it can be internal. weird though since protected is more right accessibility but we are making it internal just to hide it from public :)